pkg/app2/server.go

```diff
@@ -0,0 +1,43 @@
+package app2
```

Comment:
Should be make an additional module to put all the server-side related components? (maybe called `app/appsrv`)

Because right now, when we look at a structure, it's hard to determine whether it is used on the serverside or the clientside.

My thoughts are:
- `/app`: For client side (the actual skywire apps): `Client`, `rpcClient`, `Conn`, `Listener`, etc.
- `/app/appserver`: For server side (visor): `Networker`, `RPCGateway`, `WrappedConn`, etc.
- `/app/appnet`: For common strucutres: `Addr`, `Type`, etc.

Or something similar. What do you think? @nkryuchkov

Reply:
@evanlinjin I'll think about it

Reply:
Done partially. Still needs refactoring, moving structs around

Reply:
@Darkren is this complete now?